package/kernel/linux/modules/netdevices.mk

```diff
@@ -1441,7 +1441,7 @@ $(eval $(call KernelPackage,mlxfw))
 define KernelPackage/mlxsw-core
   SUBMENU:=$(NETWORK_DEVICES_MENU)
   TITLE:=Mellanox Technologies Switch ASICs support
-  DEPENDS:=+kmod-mlxfw +kmod-hwmon-core
+  DEPENDS:=@TARGET_x86_64 +kmod-mlxfw +kmod-hwmon-core
```

Comment:
Why do you restrict this to x86 only? Is it not working with a bigger ARM core too?

Reply:
Currently, spectrum asic is not used in devices other than x86.

Reply:
I believe so, too. I think the dependency isn't present in the Linux kernel sources because `mlxsw_core` was also used for their meanwhile EOF SwitchX-2 and SwitchIB ASICs before their drivers got removed.